include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -133,6 +133,10 @@ typedef enum OrtErrorCode {
   ORT_EP_FAIL,
 } OrtErrorCode;
 
+typedef enum OrtSessionConfigKey {
+  ORT_SESSION_CONFIG_MAXIMUM,  // Always leave this as the last value
+} OrtSessionConfigKey;
```

Comment:
Since I don't have any EP setting to add here anymore, so I will leave this as a stub
Maybe we want to add a header file for this enum with clear naming convention and string format for each config

Reply:
Could go either way on this. 
We have SessionOptions and we're adding essentially a collection of key:value pairs to that (so a collection within a collection). 
We somewhat hide the session options in parts of the API (e.g. it's SetSessionLogId not SetSessionOptionX) but make it obvious in others (CloneSessionOptions).
Due to that, OrtSessionOptionKey could potentially be confused as something to set any 'key' in SessionOptions.

Possibly the clearest is OrtSessionOptionsConfigKey. 



---
In reply to: [466683249](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466683249) [](ancestors = 466683249)

Reply:
Went with OrtSessionOptionsConfigKey

---
In reply to: [466725770](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466725770) [](ancestors = 466725770,466683249)